src/github/taskDashboardWebviewProvider.ts

```diff
@@ -0,0 +1,1180 @@
+		}
+
+		return issueData;
+	}
+
+	private async findLocalTaskBranch(branchName: string): Promise<string | undefined> {
+		try {
+			// Use the same logic as TaskManager to get all task branches
+			for (const folderManager of this._repositoriesManager.folderManagers) {
+				if (folderManager.repository.getRefs) {
+					const refs = await folderManager.repository.getRefs({ pattern: 'refs/heads/' });
+
+					// Debug: log all branches
+					Logger.debug(`All local branches: ${refs.map(r => r.name).join(', ')}`, DashboardWebviewProvider.ID);
+
+					// Filter for task branches and look for our specific branch
+					const taskBranches = refs.filter(ref =>
+						ref.name &&
+						ref.name.startsWith('task/')
+					);
+
+					Logger.debug(`Task branches: ${taskBranches.map(r => r.name).join(', ')}`, DashboardWebviewProvider.ID);
+					Logger.debug(`Looking for branch: ${branchName}`, DashboardWebviewProvider.ID);
+
+					const matchingBranch = taskBranches.find(ref => ref.name === branchName);
+
+					if (matchingBranch) {
+						Logger.debug(`Found local task branch: ${branchName}`, DashboardWebviewProvider.ID);
+						return branchName;
+					}
+				}
+			}
+			Logger.debug(`Local task branch ${branchName} not found in any repository`, DashboardWebviewProvider.ID);
+			return undefined;
+		} catch (error) {
+			Logger.debug(`Failed to find local task branch ${branchName}: ${error}`, DashboardWebviewProvider.ID);
+			return undefined;
+		}
+	}
+
+	private async findPullRequestForBranch(branchName: string): Promise<{ number: number; title: string; url: string } | undefined> {
+		try {
+			for (const folderManager of this._repositoriesManager.folderManagers) {
+				if (folderManager.gitHubRepositories.length === 0) {
+					continue;
+				}
+
+				// Try each GitHub repository in this folder manager
+				for (const githubRepository of folderManager.gitHubRepositories) {
+					try {
+						// Use the getPullRequestForBranch method to find PRs for this branch
+						const pullRequest = await githubRepository.getPullRequestForBranch(branchName, githubRepository.remote.owner);
+
+						if (pullRequest) {
+							Logger.debug(`Found PR #${pullRequest.number} for branch ${branchName}`, DashboardWebviewProvider.ID);
+							return {
+								number: pullRequest.number,
+								title: pullRequest.title,
+								url: pullRequest.html_url
+							};
+						}
+					} catch (error) {
+						Logger.debug(`Failed to find PR for branch ${branchName} in ${githubRepository.remote.owner}/${githubRepository.remote.repositoryName}: ${error}`, DashboardWebviewProvider.ID);
+						// Continue to next repository
+					}
+				}
+			}
+
+			Logger.debug(`No PR found for branch ${branchName}`, DashboardWebviewProvider.ID);
+			return undefined;
+		} catch (error) {
+			Logger.debug(`Failed to find PR for branch ${branchName}: ${error}`, DashboardWebviewProvider.ID);
+			return undefined;
+		}
+	}
+
+	private async switchToMainBranch(): Promise<void> {
```

Comment:
Can you use this?

https://github.com/microsoft/vscode-pull-request-github/blob/e5ce10696b0010b836f8a9a96a4ecd6a33295a48/src/github/folderRepositoryManager.ts#L2466-L2467

The first folderManager approach probably won't work in a multi repo workspace.